docs/root/overview.md

```diff
@@ -0,0 +1,189 @@
+# Nighthawk: architecture and key concepts
+
+## High level interaction model
+
+**Process** creates one or more **Workers**. **Worker** will run **Sequencer**,
+which in turn queries **RateLimiter** for [request-release
+timings](terminology.md#request-release-timings). When it is time to release a
+request, **BenchmarkClient** will be requested to do so by **Sequencer**.
+**BenchMarkClient** will then ask its underlying **Pool** to create a
```

Comment:
(nit) This part doesn't read well:

"to create a **StreamDecoder** release the actual request"